functions/pupil-prefs/v1.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const moment = require('moment')
+const sqlService = require('less-tedious')
+const { TYPES } = require('tedious')
+const uuid = require('uuid/v4')
+const winston = require('winston')
+
+const accessArrangementsTable = '[accessArrangements]'
+const azureStorageHelper = require('../lib/azure-storage-helper')
+const config = require('../config')
+const checkTable = '[check]'
+const pupilAccessArrangementsTable = '[pupilAccessArrangements]'
+const pupilFontSizesTable = '[pupilFontSizes]'
+const pupilColourContrastsTable = '[pupilColourContrasts]'
+const pupilTable = '[pupil]'
+const schema = '[mtc_admin]'
+
+winston.level = 'error'
+sqlService.initialise(config)
+
+const v1 = {}
+
+v1.process = async function (context, pupilPrefsMessage) {
+  const { checkCode, preferences } = pupilPrefsMessage
+  const { fontSizeCode, colourContrastCode } = preferences
+  context.log('pupil-prefs: message received', checkCode, fontSizeCode, colourContrastCode)
+
+  // Update pupil access arrangements preference based on code presence in the payload
+  try {
+    if (fontSizeCode) {
+      await v1.updatePupilAccessArrangementsPreference(checkCode, 'pupilfontSizes_id', pupilFontSizesTable, fontSizeCode, 'FTS')
+      context.log(`pupil-prefs: SUCCESS: pupil access arrangement row updated for checkCode ${checkCode} and font size code ${fontSizeCode}`)
+    }
```

Comment:
Should this be an else clause, rather than sequential `if`s?   

If I see two `if`s then I assume that both can be true in the same run, whereas, if there is an if/else construct then I know the values are exclusive.

Reply:
@jon-shipley updated

Reply:
As discussed I am undoing this since the SPA will always submit both colour contrast and font size even with their default values